ozaria/site/components/interactive/PageInteractive/insertCode/index.vue

```diff
@@ -75,32 +74,57 @@
 
         let selectedAnswerLine = ''
         if (this.selectedAnswer) {
-          selectedAnswerLine = this.selectedAnswer.line.trim()
+          selectedAnswerLine = this.selectedAnswer.text.trim()
         }
 
         return `${splitSampleCode[0]}\n${selectedAnswerLine}\n${splitSampleCode[1]}`.trim()
       },
 
       answerOptions () {
-        return [
-          { id: 1, line: 'line.one()' },
-          { id: 2, line: 'const z = "aaa"' },
-          { id: 3, line: 'line.three()' },
-          { id: 4, line: 'line.four()' }
-        ]
+        return this.localizedInteractiveConfig.choices
+      },
+
+      codemirror () {
+        return this.$refs.codeMirrorComponent.codemirror
+      }
+    },
+
+    watch: {
+      selectedAnswer () {
+        this.updateHighlightedLine()
       }
     },
 
     methods: {
       selectAnswer (answer) {
-        this.selectedAnswer = answer
+        this.selectedAnswer = {
+          ...answer,
+          triggerArt: toUriFilePath(answer.triggerArt)
+        }
+      },
+
+      onCodeMirrorReady () {
+        this.codemirrorReady = true
+        this.updateHighlightedLine()
+      },
+
+      updateHighlightedLine () {
+        // FIXME: This method doesn't seem to work at all. Line style is not applied.
```

Comment:
TODO

Reply:
This may be a race condition from when the code updates in the codemirror component to when our callbacks to update the highlighted line are fired.  May want to look into other codemirror events we can use.

Reply:
I agree that this is likely a race condition. There may be a predictable way to ensure it always executes. For example maybe it can be placed on a lifecycle hook like `updated` with a slight delay (45ms)?

Reply:
Was able to prove that it was a race condition with a debugger call, class change was performed prior to line change.  Used input event to listen for code updates so that we change classes at the proper time.